src/main/java/org/reactome/release/orthopairs/UniProtGeneNamesRetriever.java

```diff
@@ -0,0 +1,178 @@
+        // Query UniProt for gene names associated with UniProt identifiers.
+        return retrieveGeneNamesFromUniProt(partitionedUniProtIds);
+
+    }
+
+    /**
+     * Get all UniProt identifiers in pantherMappings and then partition them into chunks of 250 identifiers.
+     * @param pantherMappings Map<String, Set<String>> - Species-specific homolog mappings.
+     * @return <List><Set<String>> Partitioned UniProt identifiers.
+     */
+    private static List<Set<String>> getPartitionedUniProtIdentifiers(Map<String, Set<String>> pantherMappings) {
+
+        // Get all UniProt identifiers in the Map, removing duplicates.
+        Set<String> uniprotIds = getUniProtIdentifiers(pantherMappings);
+        logger.info("Found " + uniprotIds.size() + " UniProt accessions");
+        // Partition the UniProt identifiers into chunks of 250.
+        return partitionUniProtIdentifiers(uniprotIds);
+    }
+
+    /**
+     * Get all UniProt identifiers in pantherMappings.
+     * @param pantherMappings Map<String, Set<String>> - Species-specific homolog mappings.
+     * @return Set<String> - All UniProt identifiers in the mapping object.
+     */
+    public static Set<String> getUniProtIdentifiers(Map<String, Set<String>> pantherMappings) {
+        Set<String> uniprotIds = new HashSet<>();
+        for (String uniprotKey : pantherMappings.keySet()) {
+            for (String uniprotValue : pantherMappings.get(uniprotKey)) {
+                // Some identifiers aren't UniProt, and therefore can't be queried for Gene Names.
+                if (uniprotValue.contains("UniProtKB")) {
+                    // The values at this point are of the form "UniProtKB=123456", so we isolate the identifier.
+                    uniprotIds.add(uniprotValue.split("=")[1]);
+                }
+            }
+        }
+        return uniprotIds;
+    }
+
+    /**
+     * Partition all UniProt identifiers into 250-identifier chunks, and add each chunk to a List object.
+     * @param uniprotIds Set<String> - All unique UniProt identifiers that were in the Panther mapping.
+     * @return List<Set<String>> Partitioned UniProt identifiers.
+     */
+    public static List<Set<String>> partitionUniProtIdentifiers(Set<String> uniprotIds) {
+        List<Set<String>> partitionedUniProtIds = new ArrayList<>();
+        Set<String> partition = new HashSet<>();
+        for (String uniprotId : uniprotIds) {
+            partition.add(uniprotId);
+            // Once the partition has 250 identifiers, add it to the List and reset the partition variable.
+            if (partition.size() == MAX_UNIPROT_BATCH_QUERY_SIZE) {
+                partitionedUniProtIds.add(partition);
+                partition = new HashSet<>();
+            }
+        }
+        // The final identifiers at the end are added to the List here.
+        partitionedUniProtIds.add(partition);
+        return partitionedUniProtIds;
+    }
```

Comment:
If you are able to make the `uniprotIds` parameter into a `List` instead of a `Set`, you could do the partitioning of the chunks by using the `List` class's `subList` method:
https://stackoverflow.com/a/379824